clifford/test/test_jit_func.py

```diff
@@ -23,6 +23,31 @@ def slow_test_func(A, B):
 
         np.testing.assert_allclose(test_func(e1, e12).value, slow_test_func(e1, e12).value)
 
+    def test_nested_functions(self):
+        e1 = self.blades['e1']
+        e12 = self.blades['e12']
+
+        def test_func_1(A, B):
+            op = (A(1)+B(2))(1)
+            return op
+
+        def test_func_2(A):
+            op = ~A + 5*e12
+            return op
+
+        def compound_func(A, B):
+            return test_func_2(test_func_1(A, B))
+
+        test_func_1_jit = jit_func(self.layout)(test_func_1)
+        test_func_2_jit = jit_func(self.layout,
+                                   mv_constants={'e12': e12})(test_func_2)
+
+        test_compound_func = jit_func(self.layout,
+                                      mv_constants={'test_func_1': test_func_1_jit,
+                                                    'test_func_2': test_func_2_jit})(compound_func)
```

Comment:
This is a way to nest functions by using(abusing?) the mv_constants argument, it is clearly not a great solution but I'm not sure how to pull it off otherwise. Perhaps the decorator has knowledge of what functions are available?

Reply:
The decorator can probably use `inspect.currentframe().f_back.f_locals`

Reply:
I suspect you should be using `f_globals` instead of `globals()` too.